src/uiSelectDirective.js

```diff
@@ -132,6 +132,15 @@ uis.directive('uiSelect',
               });
           });
         }
+        
+        //Reset form based on scope event name (e.g. reset-search-input-on='SomeEventName')
+        if (angular.isDefined(attrs.resetSearchInputOn)){
+          scope.$on(attrs.resetSearchInputOn, function() {
+              $timeout(function(){
```

Comment:
This seems fishy - why not just do `scope.$apply(function() { $select.clearSearchInput(); });`?